lib/magento-ecs-cdk-stack.ts

```diff
@@ -1,16 +1,80 @@
 import * as cdk from 'aws-cdk-lib';
 import { Construct } from 'constructs';
-// import * as sqs from 'aws-cdk-lib/aws-sqs';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as ecs from 'aws-cdk-lib/aws-ecs';
+import * as ecsPatterns from 'aws-cdk-lib/aws-ecs-patterns';
 
 export class MagentoEcsCdkStack extends cdk.Stack {
   constructor(scope: Construct, id: string, props?: cdk.StackProps) {
     super(scope, id, props);
 
-    // The code that defines your stack goes here
+    // VPC creation
+    const vpc = new ec2.Vpc(this, 'MagentoVpc', {
+      maxAzs: 3,
+    });
 
-    // example resource
-    // const queue = new sqs.Queue(this, 'MagentoEcsCdkQueue', {
-    //   visibilityTimeout: cdk.Duration.seconds(300)
-    // });
+    // ECS cluster creation
+    const cluster = new ecs.Cluster(this, 'MagentoCluster', {
+      vpc,
+    });
+
+    // Task Definition
+    const taskDefinition = new ecs.FargateTaskDefinition(this, 'MagentoTaskDef', {
+      memoryLimitMiB: 2048,
+      cpu: 1024,
+    });
+
+    // Adding container for MariaDB
+    const mariadbContainer = taskDefinition.addContainer('MariaDB', {
+      image: ecs.ContainerImage.fromRegistry('docker.io/bitnami/mariadb:10.6'),
+      environment: {
+        'ALLOW_EMPTY_PASSWORD': 'yes',
+        'MARIADB_USER': 'bn_magento',
+        'MARIADB_DATABASE': 'bitnami_magento',
+      },
+      logging: ecs.LogDrivers.awsLogs({ streamPrefix: 'mariadb' }),
+    });
+
+    mariadbContainer.addPortMappings({
+      containerPort: 3306,
+    });
+
+    // Adding container for Elasticsearch
+    const elasticsearchContainer = taskDefinition.addContainer('Elasticsearch', {
+      image: ecs.ContainerImage.fromRegistry('docker.io/bitnami/elasticsearch:7'),
+      logging: ecs.LogDrivers.awsLogs({ streamPrefix: 'elasticsearch' }),
+    });
+
+    elasticsearchContainer.addPortMappings({
+      containerPort: 9200,
```

Comment:
Looks good! My only advice will be to extract port to make it input to the stack, to make the template more generic and configurable